Makefile

```diff
@@ -76,6 +76,22 @@ install: LDFLAGS += -X "github.com/cockroachdb/cockroach/util.buildDeps=$(shell
 install:
 	$(GO) install -tags '$(TAGS)' $(GOFLAGS) -ldflags '$(LDFLAGS)' -v
 
+# Build, but do not run the tests.
+# PKG is expanded and all packages are built and moved to their directory.
+# If STATIC=1, tests are statically linked.
+# eg: to statically build the sql tests, run:
+#   make testbuild PKG=./sql STATIC=1
+.PHONY: testbuild
+testbuild: TESTS := $(shell $(GO) list $(PKG))
+testbuild: GOFLAGS += -c
+testbuild:
+	for p in $(TESTS); do \
+	  NAME=$$(basename "$$p"); \
+	  OUT="$$NAME.test"; \
+	  DIR=$$($(GO) list -f {{.Dir}} ./...$$NAME); \
```

Comment:
`$p` is the package path relative to the repo root, we want either the path relative to our current directory, or the absolute path.
for example:
```
$ PKG=./storage/engine
$ p=$(go list ${PKG})name=$(basename $p)
$ dir=$(go list -f {{.Dir}} ./...${name})
$ echo $p
github.com/cockroachdb/cockroach/storage/engine
$ echo $name
engine
$ echo $dir
/home/marc/cockroach/src/github.com/cockroachdb/cockroach/storage/engine
```